### PR TITLE
docs: added checkbox page

### LINKED ISSUE #729: [Docs?]: Checkbox page Design section
### Latest version

- [X] I have checked the latest version

### Summary 💡

Initial commit for the Checkbox design guidelines

### Motivation 🔦

_No response_

### PR DESCRIPTION
**Pull request by bradryanbice**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Closes #729 

No current Checkbox design guidelines page.

## What is the new behavior?

Adds Checkbox design guidelines page.

### CODE REVIEW

File: website/docs/design/checkbox.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+---
+
+## Description
+
+A small list of multi-select options.
+
+## Anatomy/Structure
+
+- Checkbox-group
+  - Checkbox
```
Comment (Rykus0) [inline website/docs/design/checkbox.md:13]: I think we should include a label for the checkbox-group

Comment (bradryanbice) [reply website/docs/design/checkbox.md:13]: Ah yes, it has one, and helper text as well, I just missed those here. I'll add them. Thanks for the heads up.

File: website/docs/design/checkbox.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    - Label
+      - Optional flag
+
+## Usage
+
+- Use Checkbox when users can select all, one, or none of the available options.
+- Checkbox can be used as a stand-alone option, e.g. when asking a user to agree to certain services or statements.
+- Checkboxes are used as part of a checkbox-group, and have a group Label and optional helper text assigned to them.
+- If necessary, use a Legend to provide further context to multiple groups of Checkboxes or the choices available within a group.
+- By default, in most occasions, leave all Checkboxes unchecked. A scenarios where checked-by-default make sense includes viewing preselected options in a profile.
```
Comment (Rykus0) [inline website/docs/design/checkbox.md:24]:
```suggestion
- By default, in most occasions, leave all Checkboxes unchecked. A scenario where checked-by-default makes sense includes viewing preselected options in a profile.
```

File: website/docs/design/checkbox.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+- Labels always trail the Checkbox-target. For similar situations where the UI dictates a left-aligned label - and the option is not part of a larger, related list - consider using a switch.
+- Note that the label will be interactive and will trigger the checkbox as well, providing a larger tap target for the user.
+
+### Dos and don’ts
+
+- Do not use Checkboxes to navigate a user to another part of the experience. Utilize text links for navigation.
+- Do always provide an accompanying label and, in many cases, accompanying helper text as part of the checkbox-group as well.
+- Do always stack Checkboxes in a group to provide better scannability for the user.
+- Do not use Checkboxes as UI for showing or hiding content. Instead consider using a Switch or other control.
+
```
Comment (Rykus0) [inline website/docs/design/checkbox.md:37]: Don't know if it's an issue here, but a common problem I used to have was with certain content teams (especially Legal) wanting to include links in the checkbox label - which is not allowed.  Not sure if we need to mention it, but thought I'd check

Comment (bradryanbice) [reply website/docs/design/checkbox.md:37]: Good point. I suppose adding something along the lines of "Additional content requiring longer explanations, links, or other elements should be used and treated separately from the Checkbox."

Comment (bradryanbice) [reply website/docs/design/checkbox.md:38]: Updated to add a "do not" about this.

File: website/docs/design/checkbox.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  - Exceptions include acronyms (all capitals), proper nouns, or site sections (title-case).
+- Do not end in punctuation unless you are using a full sentence.
+- When using to confirm user consent, start the label with “I”, e.g. "I consent to receive email notifications from Pluralsight."
+
+### Content considerations
+
+- Checkbox labels should be clear and concise. If a full sentence is needed, keep it short (approx. 40-50 characters, or 10 words).
+- Frame labels positively, e.g. “Turn on notifications” rather than “Turn off notifications”.
+
+## Figma usage
```
Comment (caseybaggz) [inline website/docs/design/checkbox.md:75]:
```suggestion
```

File: website/docs/design/checkbox.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+## Feedback on this component
```
Comment (caseybaggz) [inline website/docs/design/checkbox.md:77]:
```suggestion
```

**Review by caseybaggz — APPROVED** [review]

**Pull request merged**